src/gitExtensionIntegration.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { RemoteSourceProvider, RemoteSource } from './typings/git';
+import { CredentialStore, GitHub } from './github/credentials';
+import { Remote } from './common/remote';
+import { Protocol } from './common/protocol';
+
+export class GithubRemoteSourceProvider implements RemoteSourceProvider {
+
+	readonly name = 'GitHub';
+	readonly supportsQuery = true;
+
+	constructor(private readonly credentialStore: CredentialStore) {
+
+	}
+
+	async getRemoteSources(query?: string): Promise<RemoteSource[]> {
+		const hub = await this.getHub();
+
+		if (!hub) {
+			throw new Error('Could not fetch repositories from GitHub.');
+		}
+
+		return [];
+	}
+
+	private async getHub(): Promise<GitHub | undefined> {
```

Comment:
@RMacfarlane It was so convoluted to get a instance of `GitHub`, having to figure out the right order to call `hasOctokit`, `getHub`, `loginWithConfirmation` and `login`. This took me wayyyy too long. Shouldn't this be isolated somewhere such that other areas of code can just invoke some sort of `getGitHub(): Promise<GitHub | undefined>` and be done with it?

Reply:
Yeah, it is very difficult to understand right now. Created an issue to track this: https://github.com/microsoft/vscode-pull-request-github/issues/1667

Reply:
Thanks, that's great!